Deployment/terraform_conditional_run_apim.ps1

```diff
@@ -0,0 +1,47 @@
+param (
```

Comment:
Maybe something to do if we get time for some project level clean-up, but this PowerShell is almost identical to another one in the project and so perhaps we should look to de-deduplicate.

Reply:
Hi Milo, the bulk of the terraform command code (for example, lines 10-43) could potentially be put into a third powershell to be executed from this file and from terraform_conditional_run.ps1 to minimise code re-use. This would leave terraform_conditional_run.ps1 and terraform_conditional_run_apim.ps1 mostly to just set up the parameters and export values into pipeline variables. (Note for future: for passing an existing param() block to a .ps1, see https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.2#psboundparameters)